### DISCUSSION TITLE
(Rust) Code action/completion to fill out fields of struct, partially completed

### DISCUSSION THREAD

**Discussion by nuke-web3** [helix-editor/helix#8262] [Q&A]
```rust
pub struct CompactMerkleMultiProof {
    pub leaf_indices: Vec<usize>,
    pub hashes: Vec<HashValue>,
}

pub fn example() {
    let something = CompactMerkleMultiPro/*<CURSOR HERE>*/;
}
```
In the above example, when the cursor is placed as stated above, you get a suggestion for _only_ completing to `CompactMerkleMultiProof` but _not_ one to fill out the fields of this struct... I would like a method there to select I want to auto populate the struct. Present behavior is I _must_ compelete the struct and append `{}` to have a chance at the lsp suggeting to fill out fields:

```rust
pub fn example() {
                 /*<Cursor on top of struct>*/
    let something = CompactMerkleMultiProof{};
}
```

Only then do I get an prompt to fill fields, that feels really round-about vs an additional option when writing out the partial name of the struct. IMHO this should be a _secondary_ option as most of the time I do just want the struct name.

### Is there some way to add this myself, or should this be a feat req. ?

**Comment by the-mikedavis** [comment]
This should be asked upstream in rust-analyzer. We don't modify / extend what the language server sends for completions
